compiler/frontend/src/org/jetbrains/kotlin/resolve/calls/tower/KotlinToResolvedCallTransformer.kt

```diff
@@ -182,9 +184,21 @@ class KotlinToResolvedCallTransformer(
     ): NewAbstractResolvedCall<D> {
         val psiKotlinCall = completedCallAtom.atom.psiKotlinCall
 
+        val candidateDescriptor = completedCallAtom.candidateDescriptor
+
+        completedCallAtom.setCandidateDescriptor(
+            candidateInterceptor.interceptResolvedCallAtomCandidate(
+                candidateDescriptor,
+                completedCallAtom,
+                trace,
+                resultSubstitutor,
+                diagnostics
+            )
+        )
+
         return if (psiKotlinCall is PSIKotlinCallForInvoke) {
-            val diagnosticsForVariableCall = if (completedCallAtom.candidateDescriptor is FunctionDescriptor) emptyList() else diagnostics
-            val diagnosticsForFunctionCall = if (completedCallAtom.candidateDescriptor is FunctionDescriptor) diagnostics else emptyList()
+            val diagnosticsForVariableCall = if (candidateDescriptor is FunctionDescriptor) emptyList() else diagnostics
```

Comment:
Here we should check actual candidate descriptor of call, so it's better to revert those two lines